src/main/scala/org/allenai/pipeline/FileItem.scala

```diff
@@ -0,0 +1,70 @@
+package org.allenai.pipeline
+
+import com.amazonaws.services.s3.{ AmazonS3Client, AmazonS3 }
+
+import scala.reflect.ClassTag
+
+import java.io.File
+
+/** File-like object that can be checked for existence / read / written.
+  * Abstracts over an absolute path and a serialization method.
+  * Examples: flat file (single file), RDD (multiple files).
+  *
+  * The content of a FileItem is unreliable in presence of write interruptions.
+  * Use a Producer[T].persisted(...) to ensure write atomicity.
+  */
+trait FileItem[T] {
```

Comment:
I think it's a pretty good idea to create a combined interface like this one that unifies `Artifact` and `ArtifactIo` I would call it `PersistedItem`